pkg/modules/promote/implpromote/module.go

```diff
@@ -0,0 +1,180 @@
+package implpromote
+
+import (
+	"context"
+	"fmt"
+	"maps"
+	"slices"
+	"strings"
+	"time"
+
+	schemamigrator "github.com/SigNoz/signoz-otel-collector/cmd/signozschemamigrator/schema_migrator"
+	"github.com/SigNoz/signoz/pkg/errors"
+	"github.com/SigNoz/signoz/pkg/modules/promote"
+	"github.com/SigNoz/signoz/pkg/telemetrylogs"
+	"github.com/SigNoz/signoz/pkg/telemetrymetadata"
+	"github.com/SigNoz/signoz/pkg/telemetrystore"
+	"github.com/SigNoz/signoz/pkg/types/promotetypes"
+)
+
+var (
+	CodeFailedToPrepareBatch       = errors.MustNewCode("failed_to_prepare_batch_promoted_paths")
+	CodeFailedToSendBatch          = errors.MustNewCode("failed_to_send_batch_promoted_paths")
+	CodeFailedToAppendPath         = errors.MustNewCode("failed_to_append_path_promoted_paths")
+	CodeFailedToCreateIndex        = errors.MustNewCode("failed_to_create_index_promoted_paths")
+	CodeFailedToQueryPromotedPaths = errors.MustNewCode("failed_to_query_promoted_paths")
+)
+
+type module struct {
+	store telemetrystore.TelemetryStore
+}
+
+func NewModule(store telemetrystore.TelemetryStore) promote.Module {
+	return &module{store: store}
+}
+
+func (m *module) ListBodySkipIndexes(ctx context.Context) ([]schemamigrator.Index, error) {
+	indexes, err := telemetrymetadata.ListLogsJSONIndexes(ctx, m.store)
```

Comment:
```suggestion
	indexes, err := telemetrymetadata.ListLogsJSONIndexes(ctx, m.store)
```

@srikanthccv shouldn't this come from the `telemetryMetaStore` injection?